source/code/plugins/collectd_lib.rb

```diff
@@ -39,11 +39,16 @@ def transform(record, hostname)
 
     #add additional meta such as ObjectName, InstanceName, etc.
     def transform_and_wrap(record, hostname)
-      if record !=nil and record["error_class"]=="Fluent::BufferQueueLimitError"
+      if record.to_s != '' and record["error_class"]=="Fluent::BufferQueueLimitError"
         OMS::Log.warn_once("Buffer Queue limit exceeded, collectD metrics not being sent to OMS")
         return
       end
 
+      if record.to_s != '' and !record.has_key?("dsnames")
+	$log.info "Invalid CollectD metric record found. Discarding data"
```

Comment:
Do you want to log this every time, or just warn_once? We don't want to fill log files ...